loopchain/blockchain/blocks/v1_0/block_factory.py

```diff
@@ -0,0 +1,134 @@
+import time
+from typing import TYPE_CHECKING, Sequence
+
+from lft.consensus.epoch import EpochPool
+from lft.consensus.messages.data import DataFactory
+
+from loopchain import configure_default as conf
+from loopchain import utils
+from loopchain.blockchain import Hash32, TransactionStatusInQueue
+from loopchain.blockchain.blocks.v1_0.block import Block, BlockHeader
+from loopchain.blockchain.blocks.v1_0.block_builder import BlockBuilder
+from loopchain.blockchain.blocks.v1_0.block_verifier import BlockVerifier
+from loopchain.blockchain.invoke_result import InvokeData, InvokePool
+from loopchain.blockchain.transactions import Transaction, TransactionVerifier, TransactionSerializer
+from loopchain.crypto.signature import Signer
+from loopchain.store.key_value_store import KeyValueStore
+
+if TYPE_CHECKING:
+    from loopchain.blockchain.votes.v1_0.vote import BlockVote
+    from loopchain.baseservice.aging_cache import AgingCache
+    from loopchain.store.key_value_store_plyvel import KeyValueStorePlyvel
+
+
+class BlockFactory(DataFactory):
+    NoneData = Hash32.empty()
+    LazyData = Hash32(bytes([255] * 32))
+
+    def __init__(self, epoch_pool_with_app, tx_queue: 'AgingCache', db: KeyValueStore, tx_versioner, invoke_pool: InvokePool, signer):
+        self._epoch_pool: EpochPool = epoch_pool_with_app
+        self._tx_versioner = tx_versioner
+
+        self._tx_queue: 'AgingCache' = tx_queue
+        self._invoke_pool: InvokePool = invoke_pool
+        self._db: 'KeyValueStorePlyvel' = db  # TODO: Will be replaced as DB Component
+        self._last_block: Block = ""  # FIXME: store it in memory or get it from db
+
+        # From BlockBuilder
+        self._signer: Signer = signer
+
+    async def create_data(self, data_number: int, prev_id: bytes, epoch_num: int, round_num: int,
+                          prev_votes: Sequence['BlockVote']) -> Block:
+        """Collect tx to make a block. invoke is not done here.
+
+        :param data_number:
+        :param prev_id:
+        :param epoch_num:
+        :param round_num:
+        :param prev_votes:
+        :return: Block
+        """
+        prev_id: Hash32
+
+        # Epoch.makeup_block
+        block_builder = BlockBuilder.new(BlockHeader.version, self._tx_versioner)
+        block_builder.fixed_timestamp = int(time.time() * 1_000_000)
+        block_builder.prev_votes = prev_votes
+
+        invoke_data: InvokeData = self._invoke_pool.prepare_invoke(epoch_num, round_num)
+        self._add_tx_to_block(block_builder, invoke_data.added_transactions)
+
+        # ConsensusSiever.__build_candidate_block
+        block_builder.height = data_number
+        block_builder.prev_hash = prev_id
+        block_builder.signer = self._signer
+
+        block_builder.validators_hash = invoke_data.reps_hash
+        block_builder.next_validators = invoke_data.next_preps
```

Comment:
I think two names seem to coexist. (validator and prep)

Reply:
We need to decide one of them.

which one do you prefer?